util/log/file.go

```diff
@@ -221,3 +291,101 @@ func GetLogReader(filename string, allowAbsolute bool) (io.ReadCloser, error) {
 	}
 	return nil, err
 }
+
+// int64sortable is required so we can sort on an int64 slice.
+type fileInfosortable []FileInfo
+
+func (a fileInfosortable) Len() int      { return len(a) }
+func (a fileInfosortable) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a fileInfosortable) Less(i, j int) bool {
+	return a[i].Details.Time.UnixNano() < a[j].Details.Time.UnixNano()
+}
+
+// selectFiles selects all log files that have an timestamp before the endTime and
+// the correct severity. It then sorts them in decreasing order, with the most
+// recent as the first one.
+func selectFiles(logFiles []FileInfo, severity Severity, endTimestamp int64) []FileInfo {
```

Comment:
That just sounds too verbose.  In most other modern languages, I could do this whole thing in a single lambda expression (or a collection of chained calls).  

logFiles.where(z => z.Details.Severity == severity).where(z => z.Details.Time.UnixNano() < endTimestamp).orderBy(z => z.Details.Time.UnixNano())

It's a tiny function and it's clear from the documentation that it sorts as well.  A select statement usually has some sorting in it.